File: Template_sast/Vuln_code/ruby/1.rb
# config/initializers/session_store.rb
Rails.application.config.session_store :cookie_store, key: '_my_app_session', expire_after: 1.year

# app/controllers/sessions_controller.rb
class SessionsController < Application_Controller
  def create
    session[:user_id] = User.find_by(email: params[:email]).id
    redirect_to dashboard_path
  end

  def destroy
    # Incorrect: Only clears session data, cookie persists
    session.delete(:user_id)
    redirect_to login_path
  end
end

File: Template_sast/Vuln_code/js/1.js
const express = require('express');
const session = require('express-session');
const app = express();

app.use(session({
  secret: 'supersecret',
  resave: false,
  saveUninitialized: true,
  cookie: { maxAge: 365 * 24 * 60 * 60 * 1000 } // 1 year
}));

app.post('/login', (req, res) => {
  req.session.userId = 123;
  res.redirect('/dashboard');
});

app.get('/logout', (req, res) => {
  // Incorrect: Only destroys session, cookie persists
  req.session.destroy();
  res.redirect('/login');
});

app.listen(3000);